### ISSUE TITLE
GLES2 - shader compiler fails with multiple macro redefinition's error

### ISSUE THREAD

**Issue by avril-gh** [godotengine/godot#38248]
**Godot version:**
Godot Engine v3.2.2.beta.custom_build.16ec167b7

**OS/device including version:**
Android 4.4, OpenGL ES 2.0 PowerVR SGX 540
```
GLES: Imagination Technologies, PowerVR SGX 540, OpenGL ES 2.0
GL_EXT_debug_marker GL_OES_rgb8_rgba8 GL_OES_depth24 GL_OES_vertex_half_float GL_OES_texture_float GL_OES_texture_half_float GL_OES_element_index_uint GL_OES_mapbuffer GL_OES_fragment_precision_high GL_OES_compressed_ETC1_RGB8_texture GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_required_internalformat GL_OES_depth_texture GL_OES_get_program_binary GL_OES_packed_depth_stencil GL_OES_standard_derivatives GL_OES_vertex_array_object GL_OES_egl_sync GL_OES_surfaceless_context GL_EXT_discard_framebuffer GL_EXT_blend_minmax GL_EXT_multi_draw_arrays GL_EXT_multisampled_render_to_texture GL_EXT_shader_texture_lod GL_EXT_texture_format_BGRA8888 GL_EXT_texture_rg GL_IMG_shader_binary GL_IMG_texture_compression_pvrtc GL_IMG_texture_npot GL_IMG_texture_format_BGRA8888 GL_IMG_read_format GL_IMG_program_binary GL_IMG_uniform_buffer_object GL_IMG_multisampled_render_to_texture GL_KHR_debug
EGL
EGL_IMG_client_api_ogl EGL_KHR_image EGL_KHR_image_base EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_vg_parent_image EGL_IMG_cl_image EGL_ANDROID_blob_cache EGL_ANDROID_image_native_buffer EGL_ANDROID_framebuffer_target EGL_ANDROID_native_fence_sync EGL_ANDROID_recordable EGL_KHR_fence_sync EGL_IMG_context_priority EGL_IMG_hibernate_process EGL_IMG_image_plane_attribs EGL_KHR_surfaceless_context EGL_KHR_wait_sync EGL_KHR_create_context

```

**Issue description:**
shader compilation fails with

```
**ERROR**: GL ERROR: Source: Shader Compiler	Type: Error	ID: 0	Severity: High	Message: ERROR: 0:6: Syntax error, 'ENABLE_TANGENT_INTERP' macro redefinition
   At: drivers\gles2\rasterizer_gles2.cpp:133:_gl_debug_print() - GL ERROR: Source: Shader Compiler	Type: Error	ID: 0	Severity: High	Message: ERROR: 0:6: Syntax error, 'ENABLE_TANGENT_INTERP' macro redefinition
**ERROR**: GL ERROR: Source: Shader Compiler	Type: Error	ID: 0	Severity: High	Message: ERROR: 0:8: Syntax error, 'LIGHT_USE_RIM' macro redefinition
   At: drivers\gles2\rasterizer_gles2.cpp:133:_gl_debug_print() - GL ERROR: Source: Shader Compiler	Type: Error	ID: 0	Severity: High	Message: ERROR: 0:8: Syntax error, 'LIGHT_USE_RIM' macro redefinition

```
```
1: #version 100
2: #define ANDROID_ENABLED
3: #define SPECULAR_BLINN
4: #define ENABLE_COLOR_INTERP
5: #define ENABLE_TANGENT_INTERP
6: #define ENABLE_TANGENT_INTERP
7: #define LIGHT_USE_RIM
8: #define LIGHT_USE_RIM
9: 
10: #ifdef USE_GLES_OVER_GL
11: #define lowp
12: #define mediump
13: #define highp
14: #else
15: precision highp float;
16: precision highp int;
17: #endif
18: 
```

```
**ERROR**: SceneShaderGLES2: Vertex shader compilation failed:
Compile failed.
ERROR: 0:11: Syntax error, 'ENABLE_TANGENT_INTERP' macro redefinition
ERROR: 0:13: Syntax error, 'LIGHT_USE_RIM' macro redefinition
ERROR: 2 compilation errors. No code generated.

   At: drivers\gles2\shader_gles2.cpp:129:_display_error_with_code() - SceneShaderGLES2: Vertex shader compilation failed:
Compile failed.
ERROR: 0:11: Syntax error, 'ENABLE_TANGENT_INTERP' macro redefinition
ERROR: 0:13: Syntax error, 'LIGHT_USE_RIM' macro redefinition
ERROR: 2 compilation errors. No code generated.

**ERROR**: Method failed. Returning: __null
   At: drivers\gles2\shader_gles2.cpp:294:get_current_version() - Method failed. Returning: __null
**ERROR**: Condition "!version" is true. Returned: false
   At: drivers\gles2\shader_gles2.cpp:88:bind() - Condition "!version" is true. Returned: false
**ERROR**: Condition "!version" is true. Returned: -1
   At: drivers\gles2/shader_gles2.h:267:_get_uniform() - Condition "!version" is true. Returned: -1
**ERROR**: Condition "!version" is true. Returned: -1
   At: drivers\gles2/shader_gles2.h:267:_get_uniform() - Condition "!version" is true. Returned: -1
**ERROR**: Condition "!version" is true. Returned: -1
   At: drivers\gles2/shader_gles2.h:267:_get_uniform() - Condition "!version" is true. Returned: -1
**ERROR**: Condition "!version" is true. Returned: -1
   At: drivers\gles2/shader_gles2.h:267:_get_uniform() - Condition "!version" is true. Returned: -1
**ERROR**: Condition "!version" is true. Returned: -1
   At: drivers\gles2/shader_gles2.h:267:_get_uniform() - Condition "!version" is true. Returned: -1
**ERROR**: Condition "!version" is true. Returned: -1
   At: drivers\gles2/shader_gles2.h:267:_get_uniform() - Condition "!version" is true. Returned: -1

```
full log - [full_log.zip](https://github.com/godotengine/godot/files/4537351/full_log.zip)

**Steps to reproduce:**
run attached project

**Important note**
I can reproduce this error only on PowerVR device.
I seems to not happen on other devices like Mali or VideoCore (or they are immune to it)

**Minimal reproduction project:**
shader in this project do really nothing, its just to trigger the error.
[PowerVR-10.zip](https://github.com/godotengine/godot/files/4537374/PowerVR-10.zip)

**Comment by clayjohn** [comment]
Ah, this looks like it happens when we have multiple usage defines that map onto the same define. For example, if the user uses ``RIM``, or ``RIM_TINT`` then ``LIGHT_USE_RIM``. However, it appears that using both ``RIM`` and ``RIM_TINT`` defines ``LIGHT_USE_RIM`` twice. So the solution is going to be checking if the define has been done before doing it again here (and in all similar blocks):

https://github.com/godotengine/godot/blob/be5a47e75d4b671a48b0a35eb17887ee38d8516c/drivers/gles2/shader_compiler_gles2.cpp#L489-L498

**Comment by akien-mga** [comment]
Fixed by #38251.

**Comment by avril-gh** [comment]
tested, can confirm things work now as should. Thanks!
